index.js

```diff
@@ -66,9 +66,9 @@ exports.Database.prototype.init = function (callback) {
   const db = new this.dbModule.Database(this.dbSettings);
   this.db = new cacheAndBufferLayer.Database(db, this.wrapperSettings, this.logger);
   if (callback) {
-    this.db.init(callback);
+    util.callbackify(this.db.init.bind(this.db))(callback);
```

Comment:
intentional switcheroo?

Reply:
Yup. The API for `init()` in `index.js` is unchanged; it's the API for `init()` in `lib/CachAndBufferLayer.js` that's changing with this PR. Thus the need for the switcheroo.